### PR TITLE
created associations

### PR DESCRIPTION
**Pull request by fasaldana**
# OOP school library: set up associations

- Created new classes: `classroom.rb` `book.rb` `rental.rb`
- Set up associations between classes

### CODE REVIEW

File: src/classroom.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Classroom
+  attr_accessor :label
+
+  def initialize(label)
+    @label = label
+  end
+
+  has_many :students
```
Comment (omar-labana) [inline src/classroom.rb]: - [ ] The has-many / belongs-to relationships between Classroom and Student should be implemented according to this [article](`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships`). Kindly take a careful look at the requirements.

File: src/student.rb [hunk 2]
```diff
@@ -9,4 +9,6 @@ def initialize(age, classroom, name = 'unknown', parent_permission: true)
   def play_hooky
     '¯\(ツ)/¯'
   end
+
+  belongs_to :classroom
```
Comment (omar-labana) [inline src/student.rb]: - [ ] The has-many / belongs-to relationships between Classroom and Student should be implemented according to this [article](`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships`). Kindly take a careful look at the requirements.

File: src/book.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+  end
+
+  has_many :rentals
```
Comment (omar-labana) [inline src/book.rb]: - [ ] the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. They should be implemented according to this [lesson](`tps://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship`). Kindly take a careful look at the requirements.

File: src/person.rb [hunk 4]
```diff
@@ -20,6 +20,8 @@ def correct_name
     @name
   end
 
+  has_many :rental
```
Comment (omar-labana) [inline src/person.rb]: - [ ] Please refer to this [lesson](`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship`) to understand how this feature should be implemented.

**Review by omar-labana — CHANGES_REQUESTED** [review]
Hi @fasaldana, 

Excellent work so far! 👌
You have implemented almost all the required features. ✔️ 
There are some cases that you need to adjust but you are almost done! 😀

## Required Changes ♻️
- [ ] It seems that you are using rails' `Active Record Associations` across our code base which wont have any effect on the classes logic and will throw runtime errors. Kindly take another look at the associations lessons and implement the required logic accordingly.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/person.rb [hunk 5]
```diff
@@ -20,6 +21,10 @@ def correct_name
     @name
   end
 
+  def add_rental
+    @rentals << rental
+  end
+
```
Comment (RolandM99) [inline src/person.rb]: - [ ] Great job by trying to implement the requested changes. But, in your implementation you missed something. In this method for adding rental the person itself should be sent as a parameter to create the new rental achieving the `'has-many'` association.
you can implement it as in the example below:
```ruby
 def add_rental(book, date)
     Rental.new(date, book, self)
  end
```

File: src/book.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals << rental
+  end
```
Comment (RolandM99) [inline src/book.rb]: - [ ] As I suggested you in the person class to implement the `add_rental` method, For this class too the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association.
take a look here :
```ruby
def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

**Review by RolandM99 — CHANGES_REQUESTED** [review]
Hi @fasaldana, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DeliceLydia — APPROVED** [review]
## Hi @fasaldana, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 



 ### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**